sse2neon.h

```diff
@@ -1834,82 +1834,115 @@ FORCE_INLINE __m128i _mm_slli_epi64(__m128i a, int imm)
         vshlq_s64(vreinterpretq_s64_m128i(a), vdupq_n_s64(imm)));
 }
 
-// Shifts the 8 signed or unsigned 16-bit integers in a right by count bits
-// while shifting in zeros.
+// Shift packed 16-bit integers in a right by imm8 while shifting in zeros, and
+// store the results in dst.
 //
-//   r0 := srl(a0, count)
-//   r1 := srl(a1, count)
-//   ...
-//   r7 := srl(a7, count)
+//   FOR j := 0 to 7
+//     i := j*16
+//     IF imm8[7:0] > 15
+//       dst[i+15:i] := 0
+//     ELSE
+//       dst[i+15:i] := ZeroExtend16(a[i+15:i] >> imm8[7:0])
+//     FI
+//   ENDFOR
 //
-// https://msdn.microsoft.com/en-us/library/6tcwd38t(v=vs.90).aspx
-#define _mm_srli_epi16(a, imm)                                   \
-    __extension__({                                              \
-        __m128i ret;                                             \
-        if ((imm) <= 0) {                                        \
-            ret = a;                                             \
-        } else if ((imm) > 31) {                                 \
-            ret = _mm_setzero_si128();                           \
-        } else {                                                 \
-            ret = vreinterpretq_m128i_u16(                       \
-                vshrq_n_u16(vreinterpretq_u16_m128i(a), (imm))); \
-        }                                                        \
-        ret;                                                     \
+// https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=_mm_srli_epi16
+#define _mm_srli_epi16(a, imm)                                             \
+    __extension__({                                                        \
+        __m128i ret;                                                       \
+        if ((imm) == 0) {                                                  \
```

Comment:
How about the following changes?
```c
if ((imm) <= 0) {
    ret = a;
} else if ((imm) > 15) {
    ret = _mm_setzero_si128();
} else {
    ret = vreinterpretq_m128i_u16(vshrq_n_u16(vshlq_u16(vreinterpretq_u16_m128i(a),
         (((imm) & 15) | (((imm) & 15) == 0))));
}
```

Reply:
As I executed the `_mm_srli_epi16` on X86, it turns out that when the `imm` is less than 0 or greater than 15, the vector element would become 0.

The reason of not using `vshrq_n_u16` is that the second argument of it should be the constant immediate value.
However, the `_mm_srli_epi16` does not.